pkg/client/typed/discovery/discovery_client.go

```diff
@@ -212,6 +212,9 @@ func setDiscoveryDefaults(config *restclient.Config) error {
 	config.APIPath = ""
 	config.GroupVersion = nil
 	config.Codec = runtime.NoopEncoder{api.Codecs.UniversalDecoder()}
+	if config.UserAgent == "" {
```

Comment:
This is fine with me but @smarterclayton would likely prefer a len 0 check :smile: